src/pocketmine/form/MenuOption.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\form;
+
+/**
+ * Represents an option on a MenuForm. The option is shown as a button and may optionally have an image next to it.
+ */
+class MenuOption implements \JsonSerializable{
+
+	/**
+	 * @var string
+	 */
+	private $text;
+	/**
+	 * @var FormIcon|null
+	 */
+	private $image;
+
+	public function __construct(string $text, ?FormIcon $image = null){
+		$this->text = $text;
+		$this->image = $image;
+	}
+
+	public function getText() : string{
+		return $this->text;
+	}
+
+	public function hasImage() : bool{
+		return $this->image !== null;
+	}
+
+	public function getImage() : ?FormIcon{
+		return $this->image;
+	}
+
+	public function jsonSerialize(){
+		$json = [
+			"text" => $this->text
+		];
+
+		if($this->hasImage()){
+			$json["image"] = $this->image;
+		}
+
+		return $json;
+	}
+
+}
```

Comment:
What about also adding an empty `public function onSelected()`, which is called before/after `MenuForm::onSubmit()` is called? This can allow developers to write their handlers in the menu options instead of from the big MenuForm object, resulting in extra routing work. Without adding this in the core, developers have to add a few lines of routing code + assertion that the `MenuOption::onSelected()` method exists in order to put the handler code with the MenuOption.

Reply:
before, or after? If a developer does it themselves, they are explicit about when they want it to be called, which can be any time - if the core does it, then developers have to get used to BEFORE, or AFTER, no in-betweens.